keel-core/src/main/kotlin/com/netflix/spinnaker/keel/api/Constraint.kt

```diff
@@ -60,6 +61,14 @@ data class ManualJudgementConstraint(
   val timeout: Duration = Duration.ofDays(7)
 ) : StatefulConstraint("manual-judgement")
 
+data class PipelineConstraint(
+  val serviceAccount: String = DEFAULT_SERVICE_ACCOUNT,
```

Comment:
I think this needs to come from the delivery config. The default service account has all the permissions, and therefore this would be a huge escalation of privilege. As a first step, can we just get any resource in the environment and take the service account from that resource instead of using the default one?

Reply:
I'd rather we move the declaration of a service account to the root level of the delivery config. I don't think it should be declared on resources; perhaps we shouldn't even allow resources to be submitted outside of a delivery config.

Reply:
Let's have the "how much do we support resources separate from delivery configs" offline.

Can require specification of the service account and not have it use the default in the meantime? Or, have the pipeline run as anonymous as the default? I just don't want there to be such an easy way to get all the privileges for all the accounts.

Reply:
How about for this PR, I remove `serviceAccount` from the constraint definition, leaving pipelines to only be runnable under the keel `DEFAULT_SERVICE_ACCOUNT`, with a TODO to use the service account defined on the delivery config once that's a requirement?

Reply:
I assumed keel's `DEFAULT_SERVICE_ACCOUNT` was heavily write restricted, and that keel was only able to actuate in restricted accounts via user specified service accounts. Sound like that isn't the case which we may want to discuss further offline but I'll make this default to "anonymous" for now.